File: PathoFroh/src/main/java/com/patho/main/model/StainingPrototypeDetails.java
package com.patho.main.model;

import com.patho.main.model.interfaces.ID;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.Date;

@Entity
@SequenceGenerator(name = "stainingPrototypeDetails_sequencegenerator", sequenceName = "stainingPrototypeDetails_sequence")
@Getter
@Setter
public class StainingPrototypeDetails implements ID, Cloneable {

	@Id
	@GeneratedValue(generator = "stainingPrototypeDetails_sequencegenerator")
	@Column(unique = true, nullable = false)
	private long id;

	@Column(columnDefinition = "VARCHAR")
	private String name;

	/**
	 * Parent
	 */
	@ManyToOne(fetch = FetchType.LAZY)
	private StainingPrototype staining;

	/**
	 * rabbit/ mouse
	 */
	@Column(columnDefinition = "VARCHAR")
	private String host;

	/**
	 * e.g. human
	 */
	@Column(columnDefinition = "VARCHAR")
	private String specifity;

	/**
	 * e.g. 2ml
	 */
	@Column(columnDefinition = "VARCHAR")
	private String quantityDelivered;

	@Column(columnDefinition = "VARCHAR")
	private String positiveControl;

	/**
	 * temperature
	 */
	@Column(columnDefinition = "VARCHAR")
	private String storage;

	@Column(columnDefinition = "VARCHAR")
	@Type(type = "date")
	private Date bestBefore;

	@Column(columnDefinition = "VARCHAR")
	@Type(type = "date")
	private Date deliveryDate;

	@Column(columnDefinition = "VARCHAR")
	@Type(type = "date")
	private Date emptyDate;

	/**
	 * e.g firm
	 */
	@Column(columnDefinition = "VARCHAR")
	private String supplier;

	@Column(columnDefinition = "VARCHAR")
	private String treatment;

	@Column
	private int incubationTime;

	@Column(columnDefinition = "VARCHAR")
	private String dilution;

	@Column(columnDefinition = "VARCHAR")
	private String standardDilution;

	@Column(columnDefinition = "VARCHAR")
	private String process;

	@Column(columnDefinition = "VARCHAR")
	private String commentary;

	public StainingPrototypeDetails() {

	}

	public StainingPrototypeDetails(StainingPrototype stainingPrototype) {
		this.staining = stainingPrototype;
	}

	public StainingPrototypeDetails clone() throws CloneNotSupportedException {
		return (StainingPrototypeDetails) super.clone();
	}
}

File: PathoFroh/src/main/kotlin/com/patho/main/model/preset/StainingPrototypeDetails.kt
package com.patho.main.model.preset

import com.patho.main.model.AbstractPersistable
import org.hibernate.annotations.SelectBeforeUpdate
import org.hibernate.annotations.Type
import java.util.*
import javax.persistence.*

@Entity
@SelectBeforeUpdate(true)
open class StainingPrototypeDetails : AbstractPersistable {

    @Id
    @GeneratedValue(generator = "stainingPrototypeDetails_sequencegenerator")
    @SequenceGenerator(name = "stainingPrototypeDetails_sequencegenerator", sequenceName = "stainingPrototypeDetails_sequence")
    @Column(unique = true, nullable = false)
    override var id: Long = 0

    @Column(columnDefinition = "VARCHAR")
    open var name: String = ""

    /**
     * Parent
     */
    @ManyToOne(fetch = FetchType.LAZY)
    open var staining: StainingPrototype

    /**
     * rabbit/ mouse
     */
    @Column(columnDefinition = "VARCHAR")
    open var host: String = ""

    /**
     * e.g. human
     */
    @Column(columnDefinition = "VARCHAR")
    open var specifity: String = ""

    /**
     * e.g. 2ml
     */
    @Column(columnDefinition = "VARCHAR")
    open var quantityDelivered: String = ""

    @Column(columnDefinition = "VARCHAR")
    open var positiveControl: String = ""

    /**
     * temperature
     */
    @Column(columnDefinition = "VARCHAR")
    open var storage: String = ""

    @Column(columnDefinition = "VARCHAR")
    @Type(type = "date")
    open var bestBefore: Date = ""

    @Column(columnDefinition = "VARCHAR")
    @Type(type = "date")
    open var deliveryDate: Date? = ""

    @Column(columnDefinition = "VARCHAR")
    @Type(type = "date")
    open var emptyDate: Date? = ""

    /**
     * e.g firm
     */
    @Column(columnDefinition = "VARCHAR")
    open var supplier: String = ""

    @Column(columnDefinition = "VARCHAR")
    open var treatment: String = ""

    @Column
    open var incubationTime = 0

    @Column(columnDefinition = "VARCHAR")
    open var dilution: String = ""

    @Column(columnDefinition = "VARCHAR")
    open var standardDilution: String = ""

    @Column(columnDefinition = "VARCHAR")
    open var process: String = ""

    @Column(columnDefinition = "VARCHAR")
    open var commentary: String = ""


    constructor(prototpye : StainingPrototype){
        this.staining = prototpye
    }
}